main.c

```diff
@@ -263,7 +264,8 @@ int WINAPI _tWinMain (HINSTANCE hThisInstance,
     exit(1);
   }
 
-  if (!IsUserAdmin() && strtod(o.ovpn_version, NULL) > 2.3 && !o.silent_connection)
+  BOOL use_iservice = (o.iservice_admin && IsWindows7OrGreater()) || !IsUserAdmin();
+  if (use_iservice && strtod(o.ovpn_version, NULL) > 2.3 && !o.silent_connection)
     CheckIServiceStatus(TRUE);
```

Comment:
Nice. This covers most cases.
If running as admin and iservice_admin is false, this won't show the "wintun will not work" warning. But I'm okay with that as this results from deliberate action by an admin.

Reply:
We could probably add a confirmation "are you sure" when user is about to unset that checkbox?